modules/fbs-core/api/src/main/scala/de/thm/ii/fbs/controller/TaskController.scala

```diff
@@ -65,7 +71,29 @@ class TaskController {
   @ResponseBody
   def getTaskResults(@PathVariable("cid") cid: Int, req: HttpServletRequest, res: HttpServletResponse): Seq[UserTaskResult] = {
     val auth = authService.authorize(req, res)
-    taskService.getTaskResults(cid, auth.id)
+    val someCourseRole = courseRegistration.getParticipants(cid).find(_.user.id == auth.id).map(_.role)
+
+    (auth.globalRole, someCourseRole) match {
+      case (GlobalRole.USER, Some(CourseRole.STUDENT)) => taskService.getTaskResults(cid, auth.id)
+        .filter(userTaskRes => isTaskPublic(cid, userTaskRes.taskID, auth))
+      case _ => taskService.getTaskResults(cid, auth.id)
+    }
+  }
+
+  /**
+    * helper function that returns false if the task is not pubic and true if it is
+    * @param cid course id
+    * @param taskId task id
+    * @param auth User
+    * @return
+    */
+  def isTaskPublic(cid: Int, taskId: Int, auth: User): Boolean = {
+    try {
+      getTaskById(cid, taskId, auth)
+      true
+    } catch {
+      case _ => false
+    }
   }
```

Comment:
I would solve the problem a bit differently and not use a `try catch` but pass a parameter to the method `getTaskById` which determines whether an exception is thrown or not.